packages/core/src/__tests__/reducer.spec.js

```diff
@@ -309,4 +309,39 @@ describe('reducer', () => {
     });
 
   });
+
+  describe('ACTION_DISPOSE handler', () => {
+    const state = {
+      SAMPLE_1: {
+        data: {
+          key: 'value of sample 1'
+        },
+      },
+      SAMPLE_2: {
+        data: {
+          key: 'value of sample 2'
+        },
+      }
+    };
+
+    const setup = (apiName) => reducer(state, { type: '@@api/DISPOSE', payload: { name: apiName } });
+
+    describe('there is data of the apiName in state', () => {
+      it('should dispose of the data of the apiName', () => {
+        expect(setup('SAMPLE_1')).to.deep.equal({
```

Comment:
if you add a referential equality assert, this will fail

```js
const nextState = setup('SAMPLE_1');
expect(nextState).to.not.equal(state); // this will fail because of `delete` keyword I explain below
```